plugin/src/main/java/io/snyk/eclipse/plugin/views/snyktoolview/SnykToolView.java

```diff
@@ -261,6 +281,75 @@ public void toggleIgnoresButtons() {
 
 	}
 
+	/*
+	 * Sets up for doing a Net New Issues scan.
+	 */
+	public void enableDelta() {
+		clearDeltaNodeListeners();
+
+		Display.getDefault().asyncExec(() -> {
+			if (this.treeViewer != null && !this.treeViewer.getTree().isDisposed()) {
+
+				TreeItem[] rootItems = getTreeViewer().getTree().getItems();
+				for (TreeItem item : rootItems) {
+					ContentRootNode node = (ContentRootNode) item.getData();
+					String projectName = node.getName();
+					String projectPath = node.getPath().toString();
+					IProject project = ResourcesPlugin.getWorkspace().getRoot().getProject(projectName);
+					String baseBranch = preferenceState.getBaseBranch(projectPath);
+					String[] localBranches = preferenceState.getLocalBranches(projectPath).toArray(String[]::new);
+
+					item.setText(String.format("%s - Click here choose base branch [ current: %s ]", projectName,
+							baseBranch));
+
+					Listener selectionListener = new Listener() {
+						@Override
+						public void handleEvent(Event event) {
+							if (event.item == item) {
+								new BaseBranchDialog().baseBranchDialog(event.display, projectPath, project,
+										localBranches);
+							}
+						}
+					};
+
+					itemListeners.put(item, selectionListener);
+
+					item.getParent().addListener(SWT.Selection, selectionListener);
+				}
+			}
+		});
+	}
+
+	/*
+	 * Disables Net New Issues scan, and starts a regular scan.
+	 */
+	public void disableDelta() {
+		clearDeltaNodeListeners();
+
+		SnykExtendedLanguageClient.getInstance().triggerScan(null);
+	}
+
+	private void clearDeltaNodeListeners() {
```

Comment:
I'm not sure, I understand why we need a listener here.

Reply:
Yes, it's the listeners on the delta nodes. In SnykToolView we have a list of listeners ```itemListeners``` that we empty when the listeners are not needed anymore.